### PR TITLE
cclient: fix threshold comparison in get_inputs

### PR DESCRIPTION
**Pull request by wusyong**
* Fix threshold comparison in get_inputs to be same as iota.js [1]
* Add final check for insufficient balance

[1] https://github.com/iotaledger/iota.js/blob/next/packages/core/src/createGetInputs.ts#L147

# Test Plan:
CI must pass

### CODE REVIEW

File: cclient/iota_client_extended_api.c [hunk 1]
```diff
@@ -138,12 +138,16 @@ retcode_t iota_client_get_inputs(iota_client_service_t const* const serv,
     // expect balance value is in order.
     CDL_FOREACH(balances_req->addresses, q_iter) {
       tmp_balance = get_balances_res_balances_at(balances_res, counter);
-      if (tmp_balance >= threshold) {
+      if (out_input->total_balance < threshold) {
         out_input->total_balance += tmp_balance;
```
Comment (jserv) [inline cclient/iota_client_extended_api.c]: How about moving `tmp_balance` to this scope?

**Comment by jserv** [comment]
Rebasing is required.

File: cclient/iota_client_extended_api.c [hunk 2]
```diff
@@ -137,13 +137,17 @@ retcode_t iota_client_get_inputs(iota_client_service_t const* const serv,
...
+      if (out_input->total_balance < threshold) {
+        tmp_balance = get_balances_res_balances_at(balances_res, counter);
         out_input->total_balance += tmp_balance;
         hash243_queue_push(&out_input->addresses, q_iter->hash);
       }
       counter++;
     }
+    // check if balance is sufficient
+    if (out_input->total_balance < threshold) {
+      ret_code = RC_CCLIENT_INSUFFICIENT_BALANCE;
```
Comment (oopsmonk) [inline cclient/iota_client_extended_api.c:149]: please add `log_warning` here.

**Review by oopsmonk — CHANGES_REQUESTED** [review]

**Review by oopsmonk — APPROVED** [review]

File: cclient/iota_client_extended_api.c [hunk 3]
```diff
@@ -138,13 +138,18 @@ retcode_t iota_client_get_inputs(iota_client_service_t const* const serv,
   if (ret_code == RC_OK && balances_req->addresses) {
     // expect balance value is in order.
     CDL_FOREACH(balances_req->addresses, q_iter) {
-      tmp_balance = get_balances_res_balances_at(balances_res, counter);
-      if (tmp_balance >= threshold) {
+      if (out_input->total_balance < threshold) {
+        tmp_balance = get_balances_res_balances_at(balances_res, counter);
```
Comment (thibault-martinez) [inline cclient/iota_client_extended_api.c]: What is the tmp for ? Can't you just directly add to `out_input->total_balance` ?

**Review by thibault-martinez — APPROVED** [review]
@wusyong Thanks for the PR !

**Pull request merged**